lnrpc/rpc.proto

```diff
@@ -3000,6 +3000,11 @@ message ListInvoiceRequest {
     */
     bool pending_only = 1;
 
+    /**
+    If set, only invoies that are expired will be returned in the response.
+    */
+    bool expired_only = 2;
```

Comment:
Isn't it enough to just be able to list all `Canceled` invoices? This includes all the expired ones an perhaps a few manually canceled invoices. The distinction between the two can also be made client-side.

Perhaps it would be nice to be able to filter on invoice state in a nice way. Otherwise the list of bool flags may grow to something that we may not like.

Reply:
I was thinking about this too but then didn't want to change the semantics of the RPC call much. If we go with state filtering, then we still either want to keep the `pending_only` flag or make the state filter accept multiple options as pending is "open or accepted".

Reply:
In any case, the existing `pending_only` bool needs to stay for one more major release. But it can be deprecated and replaced by a flagged enum

Reply:
Now thinking about this again, listing only the `canceled` invoices is probably less good than listing the `expired` ones, as there will be expired invoices that are settled. ?Users may want to filter those as well.